api/v1beta1/ibmvpc_conversion.go

```diff
@@ -73,13 +122,58 @@ func (dst *IBMVPCMachineList) ConvertFrom(srcRaw conversion.Hub) error {
 func (src *IBMVPCMachineTemplate) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1beta2.IBMVPCMachineTemplate)
 
-	return Convert_v1beta1_IBMVPCMachineTemplate_To_v1beta2_IBMVPCMachineTemplate(src, dst, nil)
+	if err := Convert_v1beta1_IBMVPCMachineTemplate_To_v1beta2_IBMVPCMachineTemplate(src, dst, nil); err != nil {
+		return err
+	}
+
+	if src.Spec.Template.Spec.Image != "" {
+		dst.Spec.Template.Spec.Image = &infrav1beta2.IBMVPCResourceReference{
+			ID: &src.Spec.Template.Spec.Image,
+		}
+	}
+
+	if src.Spec.Template.Spec.ImageName != "" {
+		dst.Spec.Template.Spec.Image = &infrav1beta2.IBMVPCResourceReference{
+			Name: &src.Spec.Template.Spec.ImageName,
+		}
+	}
+
+	for _, sshKey := range src.Spec.Template.Spec.SSHKeyNames {
+		dst.Spec.Template.Spec.SSHKeys = append(dst.Spec.Template.Spec.SSHKeys, &infrav1beta2.IBMVPCResourceReference{
+			Name: sshKey,
+		})
+	}
+
+	return nil
 }
 
 func (dst *IBMVPCMachineTemplate) ConvertFrom(srcRaw conversion.Hub) error {
 	src := srcRaw.(*infrav1beta2.IBMVPCMachineTemplate)
 
-	return Convert_v1beta2_IBMVPCMachineTemplate_To_v1beta1_IBMVPCMachineTemplate(src, dst, nil)
+	if err := Convert_v1beta2_IBMVPCMachineTemplate_To_v1beta1_IBMVPCMachineTemplate(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Preserve Hub data on down-conversion except for metadata
+	if err := utilconversion.MarshalData(src, dst); err != nil {
+		return err
+	}
+
+	if src.Spec.Template.Spec.Image != nil && src.Spec.Template.Spec.Image.ID != nil {
+		dst.Spec.Template.Spec.Image = *src.Spec.Template.Spec.Image.ID
+	}
+
+	if src.Spec.Template.Spec.Image != nil && src.Spec.Template.Spec.Image.Name != nil {
+		dst.Spec.Template.Spec.ImageName = *src.Spec.Template.Spec.Image.Name
+	}
+
+	for _, sshKey := range src.Spec.Template.Spec.SSHKeys {
+		if sshKey.Name != nil {
+			dst.Spec.Template.Spec.SSHKeyNames = append(dst.Spec.Template.Spec.SSHKeyNames, sshKey.Name)
```

Comment:
This will be handled by already added custom conversion functions, which will be called implicitly. Tested.
`Convert_Slice_Pointer_v1beta2_IBMVPCResourceReference_To_Slice_Pointer_string`